src/content/reference/cluster-upgrades/index.md

```diff
@@ -0,0 +1,190 @@
+
+The master node re-creation is started first. Meanhile, the recreation of worker nodes starts, where the all worker nodes are recreated in batches. During the upgrade, **up to 33 percent of the worker nodes can be unavailable**.
+
+After recreation, worker nodes are **not expected to have the names** they had before.
+
+In process of the worker node recreation, any data stored in worker node's local storage - i. e. outside of EBS volumes - is lost.
+
+**Note:** We are in the process of making upgrades on AWS less disruptive.
+One goal is to make new nodes available first, then drain and remove old nodes.
+
+### Provider-specific details for Azure
+
+Our `azure-operator` manages tenant cluster on Azure via Azure Resource Manager (ARM) templates and Virtual Machine Scale Sets (VMSS).
+
+In an upgrade, all Virtual Machines (VM) are updates by reimaging with a new OS image and then rebooting.
+The master node is rebooted first.
+Worker nodes are then updated one by one, where each node is first drained (Pods re-scheduled to other nodes) and then reimaged and rebooted.
+
+On Azure, the node names visible to Kubernetes (e. g. `kubectl get nodes`) are not changed in an upgrade.
+
+**Note:** Any local storage in the node is erased. Persistent volumes are not affected by that.
+
+### Provider-specific details for KVM
+
+In a KVM-based cluster, each tenant cluster node consists of a Deployment and Pod in the control plane.
+In an upgrade, each of these deployments is updated after the according node has been drained, one after another, starting with the master node.
+This leads to removeal and reacreation of the Pods.
+
+## How to prepare your workloads {#recommendations}
+
+The following recommendations should help you harden your workload deployments for a tpyical upgrade process. Furthermore, they also make your workloads more resilient against unpredicted failures, high load, and resource pressure in your cluster.
+
+### Scale up and distribute workloads
+
+Make sure you have 2 or more replicas for all your deployments. For critical components you might want to go with more than just the bare minimum of 2.
+
+You can adjust this depending on your environments, e.g. running 2-3 replicas in `DEV`, and 5 or more in `PROD`.
+
+In case you are using Horizontal Pod Autoscaler, above recommendations should depict your minimum replica setting.
+
+Additionally, you should make sure your replicas do not land on the same node using [inter-pod anti-affinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#inter-pod-affinity-and-anti-affinity-beta-feature).
+
+We recommend always using soft anti-affinity to avoid exclusivity of nodes. However, this can in cases to less distirbuted workloads on resource pressure or node failure, which will need a rescheduling to be balanced again.
```

Comment:
`this can in cases to less distirbuted` should be rewritten.

Reply:
should be `this can, in cases, lead to less`